SYCL/ESIMD/api/functional/functions/functions_1d_select.hpp

```diff
@@ -0,0 +1,281 @@
+  }
+};
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename TestCaseT, typename NumSelectedElemsT, typename StrideT,
+          typename OffsetT, typename DataT, typename DimT>
+class run_test {
+  static constexpr int NumElems = DimT::value;
+  static constexpr int NumSelectedElems = NumSelectedElemsT::value;
+  static constexpr int Stride = StrideT::value;
+  static constexpr int Offset = OffsetT::value;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    static_assert(NumElems >= NumSelectedElems * Stride + Offset &&
+                  "Offset should be less than number selected elements.");
+    bool passed = true;
+    size_t alignment_value = alignof(DataT);
+
+    constexpr size_t value_for_increase_ref_data_for_fill = 50;
+    static_assert(std::numeric_limits<signed char>::max() >
+                      value_for_increase_ref_data_for_fill + NumElems,
+                  "Value that used for increase ref data for fill plus simd "
+                  "size should be less than char max value.");
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> initial_ref_data(NumElems, allocator);
+    shared_vector<DataT> ref_data_for_fill(NumElems, allocator);
+
+    shared_element<bool> is_correct_type(queue, true);
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      initial_ref_data[i] = i + 1;
+    }
+    // We should have different values in the first reference data and in the
+    // second reference data.
+    for (size_t i = 0; i < NumElems; ++i) {
+      ref_data_for_fill[i] =
+          initial_ref_data[i] + value_for_increase_ref_data_for_fill;
+    }
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *init_ref_ptr = initial_ref_data.data();
+      DataT *ref_data_for_fill_ptr = ref_data_for_fill.data();
+      DataT *const out_ptr = result.data();
+      auto is_correct_type_ptr = is_correct_type.data();
+
+      cgh.single_task<
+          Kernel<DataT, NumElems, TestCaseT, NumSelectedElemsT, StrideT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            *is_correct_type_ptr =
+                TestCaseT::template call_operator<DataT, NumElems,
+                                                  NumSelectedElems, Stride>(
+                    init_ref_ptr, ref_data_for_fill_ptr, out_ptr, Offset);
+          });
+    });
+    queue.wait_and_throw();
+
+    std::vector<size_t> selected_indexsess;
+    // Collect the indexess that has been selected.
+    for (size_t i = Offset; i < Stride * NumSelectedElems + Offset;
+         i += Stride) {
+      selected_indexsess.push_back(i);
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexsess.push_back(std::numeric_limits<size_t>::max());
+    auto selected_indexsess_ptr = selected_indexsess.begin();
```

Comment:
```suggestion
    auto next_selected_index = selected_indexes.begin();
```
It's iterator after all

Reply:
the comment has been applied [here](https://github.com/intel/llvm-test-suite/pull/875/commits/03d94c35b4a95e3ce26c99e1197bc50ccdc652aa#diff-295141c99d991cbb4971abc3bd754b2f17400cade3b579ac7bc36b28a1450dc3R178)